ext/src/zpages/tracez_processor.cc

```diff
@@ -0,0 +1,38 @@
+#include "opentelemetry/ext/zpages/tracez_processor.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace ext {
+namespace zpages {
+
+  void TracezSpanProcessor::OnStart(opentelemetry::sdk::trace::Recordable &span) noexcept {
+    std::lock_guard<std::mutex> lock(mtx_);
```

Comment:
Thanks for adding this lock. Please make sure to add a note (or issue) about this, so we don't forget to revisit this during performance tests. We can possibly us the [lock free circular buffer](https://github.com/open-telemetry/opentelemetry-cpp/pull/52) to hold running spans (and another one of those for ended ones).

This would also help us from possibly overflowing memory if the aggregator doesn't consume spans at a sustainable rate.

Reply:
Hi @pyohannes, I am working on the aggregator at the moment(will have a PR after this one). I am curious to know what you think is a sustainable rate of consumption of spans

Reply:
I'll create an issue with the points you mentioned!

Reply:
@kmanghat This depends on the number of spans the user creates. With sustainable I mean: the aggregator is able to consume all spans that are created.

Those are purely hypothetical numbers, but imagine a span is created every `1ms`, but the aggregator needs `2ms` to process one span. I think we need an upper limit of spans we cache in the processor to avoid running out of memory in this scenario. The lock free circular buffer provides that functionality.